src/midi/midi_fluidsynth.cpp

```diff
@@ -0,0 +1,245 @@
+
+void init_fluid_dosbox_settings(Section_prop &secprop)
+{
+	constexpr auto when_idle = Property::Changeable::WhenIdle;
+
+	auto *str_prop = secprop.Add_string("fluid.soundfont", when_idle, "");
+	str_prop->Set_help("Soundfont to use with FluidSynth."
+	                   "One must be specified.");
+
+	auto *int_prop = secprop.Add_int("fluid.samplerate", when_idle, 44100);
+	int_prop->SetMinMax(8000, 96000);
+	int_prop->Set_help("Sample rate (Hz) Fluidsynth will render at. (min "
+	                   "8000, max 96000");
+
+	str_prop = secprop.Add_string("fluid.gain", when_idle, "0.4");
+	str_prop->Set_help("Fluidsynth volume gain. (min 0.0, max 10.0");
+
+	int_prop = secprop.Add_int("fluid.polyphony", when_idle, 256);
+	int_prop->Set_help("Fluidsynth polyphony.");
+
+	int_prop = secprop.Add_int("fluid.cores", when_idle, 1);
+	int_prop->SetMinMax(1, 256);
+	int_prop->Set_help("Number of CPU cores Fluidsynth will use."
+	                   "(min 1, max 256)");
+
+	auto *bool_prop = secprop.Add_bool("fluid.reverb", when_idle, true);
+	bool_prop->Set_help("Enable reverb.");
+
+	bool_prop = secprop.Add_bool("fluid.chorus", when_idle, true);
+	bool_prop->Set_help("Fluidsynth chorus.");
+
+	str_prop = secprop.Add_string("fluid.reverb.roomsize", when_idle, "0.2");
+	str_prop->Set_help("Fluidsynth reverb room size.");
+
+	str_prop = secprop.Add_string("fluid.reverb.damping", when_idle, "0.0");
+	str_prop->Set_help("Fluidsynth reverb damping.");
+
+	str_prop = secprop.Add_string("fluid.reverb.width", when_idle, "0.5");
+	str_prop->Set_help("Fluidsynth reverb width.");
+
+	str_prop = secprop.Add_string("fluid.reverb.level", when_idle, "0.9");
+	str_prop->Set_help("Fluidsynth reverb level.");
+
+	int_prop = secprop.Add_int("fluid.chorus.number", when_idle, 3);
+	int_prop->Set_help("Fluidsynth chorus voices");
+
+	str_prop = secprop.Add_string("fluid.chorus.level", when_idle, "2.0");
+	str_prop->Set_help("Fluidsynth chorus level.");
+
+	str_prop = secprop.Add_string("fluid.chorus.speed", when_idle, "0.3");
+	str_prop->Set_help("Fluidsynth chorus speed.");
+
+	str_prop = secprop.Add_string("fluid.chorus.depth", when_idle, "8.0");
+	str_prop->Set_help("Fluidsynth chorus depth.");
+}
+
+bool MidiHandlerFluidsynth::Open(MAYBE_UNUSED const char *conf)
+{
+	LocaleGuard locale_guard;
+
+	Close();
+
+	auto *section = static_cast<Section_prop *>(control->GetSection("midi"));
+
+	fluid_settings_ptr_t settings(new_fluid_settings(), delete_fluid_settings);
+
+	auto get_double = [section](const char *const propname) {
+		try {
+			return std::stod(section->Get_string(propname));
+		} catch (const std::exception &e) {
+			/*
+			log_cb(RETRO_LOG_WARN,
+			       "[dosbox] error reading floating point '%s' "
+			       "conf setting: %s\n",
+			       e.what());
+		       */
+			return 0.0;
+		}
+	};
+
+	fluid_settings_setnum(settings.get(), "synth.sample-rate",
+	                      section->Get_int("fluid.samplerate"));
```

Comment:
If this is where we're telling Fluid what rate it should use; I think we can use `mixer.freq`, or if there's a config race-condition, we could get it from the Mixer's config `section->Get_int("rate");`.

Reply:
You meant `mixer.rate`, right? :)